src/components/signup/SignupState.js

```diff
@@ -130,24 +130,13 @@ const Signup = ({ navigation, screenProps }: { navigation: any, screenProps: any
 
       case 'goToPhone':
         if (w3User.image) {
-          userScreenData.avatar = await new Promise((resolve, reject) => {
-            const xmlHTTP = new XMLHttpRequest()
-
-            xmlHTTP.open('GET', w3User.image, true)
-            xmlHTTP.responseType = 'arraybuffer'
-            xmlHTTP.onload = function(e) {
-              const arr = new Uint8Array(this.response)
-              const raw = String.fromCharCode.apply(null, arr)
-              const b64 = btoa(raw)
-              const dataURL = 'data:image/png;base64,' + b64
-
-              resolve(dataURL)
-            }
-
-            xmlHTTP.onerror = reject
-
-            xmlHTTP.send()
-          })
+          userScreenData.avatar = await API.getArrayBufferFromImageUrl(w3User.image)
+            .then(response => {
+              return Buffer.from(response, 'binary').toString('base64')
```

Comment:
do we need data:image here so profile image is displayed correctly on UI?
please test that the complete flow works

Reply:
I have double-checked how the avatar field saved in profile. There is a base 64 string.

Reply:
i ment you are missing the "data:image" prefix.
You should test the code also works, not just verify the code.